apps/gateway/test/champions_test.exs

```diff
@@ -300,6 +301,114 @@ defmodule Gateway.Test.Champions do
+
+      # Check that the gold and gems afk rewards rates are now greater than the initial rewards
+      gold_currency_id = Currencies.get_currency_by_name!("Gold").id
+      gems_currency_id = Currencies.get_currency_by_name!("Gems").id
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gold_currency_id && rate.rate > 0
+             end)
+
+      assert Enum.any?(advanced_user.afk_reward_rates, fn rate ->
+               rate.currency_id == gems_currency_id && rate.rate > 0
+             end)
+
+      # Wait for the afk rewards to increase
+      :timer.sleep(2000)
+
+      # Claim afk rewards
+      SocketTester.claim_afk_rewards(socket_tester, advanced_user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = claimed_user}}
+
+      # Check that the user has received gold and gems
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gold_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gold"))
+               initial_gold = user_initial_currencies |> Enum.find(&(&1.currency.name == "Gold"))
+
+               user_gold_currency.amount > initial_gold.amount
+             end)
+
+      assert Enum.any?(claimed_user.currencies, fn currency ->
+               user_gems_currency = Enum.find(claimed_user.currencies, &(&1.currency.name == "Gems"))
+               initial_gems = user_initial_currencies |> Enum.find(&(&1.currency.name == "Gems"))
+
+               user_gems_currency.amount > initial_gems.amount
+             end)
+
+      # TODO: check that the afk rewards rates have been reset after [CHoM-380] is solved
+
+      # Play another level to increment the afk rewards rates again
+      [campaign_progression | _] = advanced_user.campaign_progresses
+      next_level_id = campaign_progression.level_id
+      SocketTester.fight_level(socket_tester, advanced_user.id, next_level_id)
+      fetch_last_message(socket_tester)
+
+      assert_receive %WebSocketResponse{
+        response_type: {:battle_result, _ = battle_result}
+      }
+
+      assert battle_result.result == "win"
+
+      # Get new user
+      SocketTester.get_user(socket_tester, user.id)
+      fetch_last_message(socket_tester)
+      assert_receive %WebSocketResponse{response_type: {:user, %User{} = more_advanced_user}}
+
+      # Check that the gold and gems afk rewards rates are now greater than the rewards before the second battle
+      gold_currency_id = Currencies.get_currency_by_name!("Gold").id
+      gems_currency_id = Currencies.get_currency_by_name!("Gems").id
+
+      assert Enum.any?(more_advanced_user.afk_reward_rates, fn rate ->
+               previous_rate = Enum.find(advanced_user.afk_reward_rates, &(&1.currency_id == gold_currency_id)).rate
+               rate.currency_id == gold_currency_id && rate.rate > previous_rate
```

Comment:
I wouldn't compare against the previous rate here, I'd rather check that the new rate is the one set on the level rewards